pkg/policy/api/selector.go

```diff
@@ -232,20 +271,23 @@ func (n *EndpointSelector) AddMatch(key, value string) {
 		n.MatchLabels = map[string]string{}
 	}
 	n.MatchLabels[key] = value
+	n.requirements = labelSelectorToRequirements(n.LabelSelector)
 }
 
 // Matches returns true if the endpoint selector Matches the `lblsToMatch`.
 // Returns always true if the endpoint selector contains the reserved label for
 // "all".
 func (n *EndpointSelector) Matches(lblsToMatch k8sLbls.Labels) bool {
-	lbSelector, err := metav1.LabelSelectorAsSelector(n.LabelSelector)
-	if err != nil {
-		// FIXME: Omit this error or throw it to the caller?
-		// We are doing the verification in the ParseEndpointSelector but
-		// don't make sure the user can modify the current labels.
-		log.WithError(err).WithField(logfields.EndpointLabelSelector,
-			logfields.Repr(n)).Error("unable to match label selector in selector")
-		return false
+	if n.requirements == nil {
```

Comment:
Why wouldn't you compute the requirements and set them into `n.requirements` if it's `nil`?
So the next time `Matches` is called, you can use the cached requirements and you'd pay the cost only once.

Reply:
Done.